pennylane/transforms/metric_tensor.py

```diff
@@ -197,7 +267,19 @@ def qnode_execution_wrapper(self, qnode, targs, tkwargs):
     cjac_fn = qml.transforms.classical_jacobian(qnode, expand_fn=_expand_fn)
 
     def wrapper(*args, **kwargs):
-        mt = mt_fn(*args, **kwargs)
+        try:
+            mt = mt_fn(*args, **kwargs)
+        except qml.wires.WireError as e:
+            warnings.warn(
+                "A wire was not found while executing tapes for the metric tensor. "
+                "Probably you tried to use Hadamard tests and the device does not provide an "
+                "additional wire or the requested auxiliary wire does not exist on the device. "
+                "Reverting to the block-diagonal approximation. It will often be much more "
+                "efficient to request the block-diagonal approximation directly! "
+                f"Warning raised from WireError:\n{e}"
+            )
+            tkwargs["approx"] = "block-diag"
+            return self.__call__(qnode, targs, tkwargs)

```

Comment:
This doesn't seem to work for me 🤔 

```
/home/josh/xanadu/pennylane/pennylane/transforms/metric_tensor.py:273: UserWarning: A wire was not found while executing tapes for the metric tensor. Probably you tried to use Hadamard tests and the device does not provide an additional wire or the requested auxiliary wire does not exist on the device. Reverting to the block-diagonal approximation. It will often be much more efficient to request the block-diagonal approximation directly! Warning raised from WireError:
Did not find some of the wires (2,) on device with wires (0, 1).
  warnings.warn(
<function circuit at 0x7f08413cb4c0>
```

Reply:
Should this be?
```suggestion
            tkwargs["approx"] = "block-diag"
            return self.qnode_wrapper(qnode, targs, tkwargs)(*args, **kwargs)
```

(note: I tried `self.__call__(...)(*args, **kwargs)`, and it also works, _but_ I get a tonne of extra warnings that I don't get with the above)

Reply:
will look into this :)